src/org/thoughtcrime/securesms/ConversationListItem.java

```diff
@@ -165,7 +165,16 @@ public void bind(@NonNull ThreadRecord thread,
     setBatchState(batchMode);
     setBgColor(thread);
 
-    this.contactPhotoImage.setAvatar(glideRequests, recipient, false);
+    this.avatar.setAvatar(glideRequests, recipient, false);
+
+    if (recipient.isMultiUserRecipient()) {
+      avatar.setStatusEnabled(false);
+    } else {
+      DcContext dcContext = DcHelper.getContext(getContext());
+      int[] members = dcContext.getChatContacts((int)chatId);
+      DcContact contact = dcContext.getContact(members.length>=1? members[0] : 0);
+      avatar.setStatusEnabled(contact.isOnline());
```

Comment:
during creation of the recipient object using `new Recipient(dcChat, null, null)` [quite the same is done](https://github.com/deltachat/deltachat-android/blob/master/src/org/thoughtcrime/securesms/recipients/Recipient.java#L124) (calling `getChatContacts()` and then `getContact()`) to figure out whether to check for a system contact photo.

we could (1) cache the contact object there (we should take care to not mess with contact-recipients then) or we could (2) offer a `Recipient.isSeenRecently()`

Reply:
yes would be nice to optimize in another PR, I limited to imitate what is done in the code-base, but even then I have not seen any performance issue in DeltaLab using this for quite a long time already, probably because it is done only for the current displayed items that are 1:1 chats (since the RecycleView thingy will not load the full list)

Reply:
it turned out that the `Recipient` class already performs this and was created in the same block of code, so I just reused that (so the performance is exactly the same as currently)

Reply:
are you sure, `Recipient.getDcContact()` works as expected? for me, the indocator is no longer shown in the chatlist with the latest commit.

reason seems to be that `Recipient.getDcContact()` currently only returns an object if the recipient is a _contact_. the recipient in question here is a _chat_, however (this is weird stuff, but that's how it is).

Reply:
oops!!! overlooked that! was thinking that in the block on `Recipient,java:121` the dcContact var was initialized but just checked and in fact it is not set! will fix, because all the logic is there just that the contact is not saved